drivers/cyw43/cyw43_lwip.c

```diff
@@ -117,7 +117,10 @@ void cyw43_tcpip_init(cyw43_t *self, int itf) {
     #else
     netif_add(n, &ipconfig[0], &ipconfig[1], &ipconfig[2], self, cyw43_netif_init, netif_input);
     #endif
-    netif_set_hostname(n, "PYBD");
+    if (self->hostname[0] == 0) {
+        strcpy(self->hostname, "PYBD");
+    }
```

Comment:
It might be worth pre-populating `self->hostname` with the default "PYBD" in `cyw43_init` and then you don't need these 3 lines.

Also would be good to make the default configurable by the board, eg `MICROPY_BOARD_HOSTNAME`.